tatorte_classifier/api/texts.py

```diff
@@ -0,0 +1,139 @@
+
+from werkzeug.exceptions import BadRequest
+
+from tatorte_classifier.database import (
+    get_all_texts,
+    get_text,
+    modify_text,
+    create_text,
+    get_random_text,
+    delete_text,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Text(Resource):
+    def get(self, text_id):
+        """
+        Input:
+            text_id: [id of text, 'random']
+        Returns:
+            {
+                "_id": {
+                    "$oid": "5c6c1b2573cda500b254404c"
+                },
+                "data": "This is a test. Number 2",
+                "time_created": "2019-02-19 15:05:09",
+                "time_modified": "2019-02-19 15:18:53",
+                "categories": [4, 2]
+            }
+        """
+        try:
+            if text_id == "random":
+                return dumps(get_random_text())
+            return dumps(get_text(text_id))
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+    def delete(self, text_id):
+        """Deletes text document with id == <text_id>
+        """
+
+        try:
+            delete_text(text_id)
+            return jsonify({"success": True})
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+    def patch(self, text_id):
+        """
+        Input:
+            {
+                "categories": [3, 2],
+            }
+
+        Returns:
+            <id>
+        """
+        try:
+            request_json = request.get_json()
+            modify_text(text_id, request_json["categories"])
+            return jsonify({"success": True})
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+
+class Texts(Resource):
+    def get(self):
+        """
+        Returns:
+            [
+                {
+                    "_id": {
+                        "$oid": "5c6c1b2573cda500b254404c"
+                    },
+                    "data": "This is a test. Number 2",
+                    "time_created": "2019-02-19 15:05:09",
+                    "time_modified": "2019-02-19 15:18:53",
+                    "categories": [4, 2]
+                }, ...
+            ]
+        """
+        try:
+            return dumps(get_all_texts())
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+    def post(self):
+        """
+        Input:
+            {
+                "data": "This is a test",
+                "categories": [3, 2]
+            }
+
+        Returns:
+            <id>
+        """
+        try:
+            request_json = request.get_json()
+            create_text(request_json["data"], request_json["categories"])
+            return jsonify({"success": True})
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+
+class TextsStartEnd(Resource):
```

Comment:
Nonono, please get rid of the basdard-resource ;)

`start`/`end` should be implemented as query parameters of the texts list endpoint and subsumed under the corresponding resource `TextList`. Otherwise you will have a hard time if you would like to extend the API with more filter options (i.e. filter for category, creation_time, etc.) because right now it's not clear where to add this functionality as you have two seperate resources representing semantically the same thing: a list of texts. Also consider, that the current implementation requires strict order of the parameters and would not be able to parse `/texts/end=...&start=...`.

For the rescue: Use `/texts/?` to mark any `key=value` pair as a query argument. That way you can filter texts by offsets, page sizes, categories and what not all on the same one endpoint all treated equally with no strict order required and only need one resource: `TextList`.

Access the query_string: https://stackoverflow.com/a/11781872

Maybe this is also a good lead:
https://flask-restful.readthedocs.io/en/latest/quickstart.html#argument-parsing